compiler/tests/generated/generic_service_grpc.h

```diff
@@ -35,7 +35,8 @@
 namespace tests
 {
 
-class Foo final
+template <typename Payload>
+    class Foo final
 {
 public:
     template <typename TThreadPool>
```

Comment:
Assuming we do not want to waste any CPU cycles, then function object should be better, unless it adds significant complexity that you want to avoid. The `std::function` comes with a cost.